### DISCUSSION TITLE
Idea: libsqlite3.so compatible wrapper loaded with LD_PRELOAD

### DISCUSSION THREAD

**Discussion by orent** [canonical/dqlite#601] [Ideas]
A library providing a subset of the libsqlite3.so API could be loaded with LD_PRELOAD and enable existing programs to use dqlite transparently, as long as they do not link sqlite statically and should include all scripting languages. It should also be useful with the sqlite3 utility, but it would need to be compiled differently for this to work.

The wrapper could also dynamically load the existing libsqlite3.so and forward calls to it for opening existing sqlite3 files. Connecting to a dqlite database could be done by opening a small stub file containing the cluster configuration or using dqlite: uri

**Comment by stgraber** [comment]
That may be possible once we have #29.

**Comment by freeekanayaka** [comment]
I assume @stgraber meant #291.

**Comment by stgraber** [comment]
Indeed :)
